src/file_analysis/File.h

```diff
@@ -176,6 +176,15 @@ class File {
      */
     void FileEvent(EventHandlerPtr h, Args args);
 
+    /**
+     * Drain the event queue and this file's analyzer modifications after
+     * raising a custom event via FileEvent.
+     *
+     * Using internal events (file_new, file_over_new_connection, file_sniff
+     * and file_extraction_limit) invokes this automatically.
+     */
+    void FileEventDrain();
```

Comment:
The method name here feels misleading: it's draining all events. Plus, "FileEvent" is used differently in the other methods just above.